tracing-appender/src/rolling.rs

```diff
@@ -469,17 +477,43 @@ impl io::Write for RollingWriter<'_> {
 // === impl Inner ===
 
 impl Inner {
-    fn refresh_writer(&self, now: OffsetDateTime, file: &mut File) {
+    #[cfg(feature = "compression_gzip")]
+    fn refresh_writer(&self, now: OffsetDateTime, file: &mut WriterChannel) {
+        debug_assert!(self.should_rollover(now));
+
+        let filename = self
+            .rotation
+            .join_date(&self.log_filename_prefix, &now, false);
+
+        let writer = if let Some(compression) = self.compression.clone() {
+            WriterChannel::new_with_compression(&self.log_directory, &filename, compression)
+        } else {
+            WriterChannel::new_without_compression(&self.log_directory, &filename)
+        };
+
+        Self::refresh_writer_channel(file, writer);
+    }
+
+    #[cfg(not(feature = "compression_gzip"))]
+    fn refresh_writer(&self, now: OffsetDateTime, file: &mut WriterChannel) {
         debug_assert!(self.should_rollover(now));
 
-        let filename = self.rotation.join_date(&self.log_filename_prefix, &now);
+        let filename = self
+            .rotation
+            .join_date(&self.log_filename_prefix, &now, false);
+
+        let writer = WriterChannel::new_without_compression(&self.log_directory, &filename);
 
-        match create_writer(&self.log_directory, &filename) {
-            Ok(new_file) => {
+        Self::refresh_writer_channel(file, writer)
+    }
```

Comment:
nit, take it or leave it: i think we can reduce the amount of duplicated code here by adding the `#[cfg]` attribute to _just_ the code that's different based on whether or not the feature flag is enabled, like this:

```rust
    fn refresh_writer(&self, now: OffsetDateTime, file: &mut WriterChannel) {
        debug_assert!(self.should_rollover(now));

        let filename = self
            .rotation
            .join_date(&self.log_filename_prefix, &now, false);
 
        #[cfg(feature = "compression_gzip")]
        let writer = if let Some(compression) = self.compression.clone() {
            WriterChannel::new_with_compression(&self.log_directory, &filename, compression)
        } else {
            WriterChannel::new_without_compression(&self.log_directory, &filename)
        };

        #[cfg(not(feature = "compression_gzip"))]
        let writer = WriterChannel::new_without_compression(&self.log_directory, &filename);

        Self::refresh_writer_channel(file, writer);
    }
```

Reply:
actually, upon looking at it closer, I noticed there's also a `WriterChannel::new` which we could just pass the compression parameter to, rather than using an `if let` here. I think this can be simplified further, to just:
```rust
    fn refresh_writer(&self, now: OffsetDateTime, file: &mut WriterChannel) {
        debug_assert!(self.should_rollover(now));

        let filename = self
            .rotation
            .join_date(&self.log_filename_prefix, &now, false);
 
        #[cfg(feature = "compression_gzip")]
        let writer = WriterChannel::new(&self.log_directory, &filename, compression, &self.compression.clone());

        #[cfg(not(feature = "compression_gzip"))]
        let writer = WriterChannel::new(&self.log_directory, &filename);

        Self::refresh_writer_channel(file, writer);
    }
```

Reply:
Yeah, this looks much cleaner indeed. Fixed.